hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -653,36 +653,44 @@ extends Importer:
       case (tree @ LetLike(`let`, lhs, S(rhs), N)) :: sts =>
         raise(ErrorReport(msg"Unsupported let binding shape" -> tree.toLoc :: Nil))
         go(sts, Nil, Term.Error :: acc)
-      case (hd @ Handle(id: Ident, cls: Ident, Block(sts_), N)) :: sts =>
+      case (hd @ Hndl(id: Ident, cls: Ident, Block(sts_), N)) :: sts =>
         reportUnusedAnnotations
-        val sym = fieldOrVarSym(HandlerBind, id)
-        log(s"Processing `handle` statement $id (${sym}) ${ctx.outer}")
-        
-        val elabed = block(sts_)._1
-        
-        elabed.res match
-          case Term.Lit(UnitLit(true)) => 
-          case trm => raise(WarningReport(msg"Terms in handler block do nothing" -> trm.toLoc :: Nil))
-        
-        val tds = elabed.stats.map {
-          case td @ TermDefinition(owner, Fun, sym, params, sign, body, resSym, flags, annotations) =>
-            params.reverse match
-              case ParamList(_, value :: Nil, _) :: newParams =>
-                val newTd = TermDefinition(owner, Fun, sym, newParams.reverse, sign, body, resSym, flags, annotations)
-                S(HandlerTermDefinition(value.sym, newTd))
-              case _ => 
-                raise(ErrorReport(msg"Handler function is missing resumption parameter" -> td.toLoc :: Nil))
-                None
-              
-          case st => 
-            raise(ErrorReport(msg"Only function definitions are allowed in handler blocks" -> st.toLoc :: Nil))
-            None
-        }.collect { case Some(x) => x }
+        val res: Term.Blk = ctx.nest(N).givenIn:
+          val sym = fieldOrVarSym(HandlerBind, id)
+          log(s"Processing `handle` statement $id (${sym}) ${ctx.outer}")
+          
+          // TODO: shouldn't need uid here
```

Comment:
Indeed, it shouldn't be needed. We'll be able to remove it when I change my symbols change.